sdk/trace/provider.go

```diff
@@ -239,10 +254,10 @@ func (p *TracerProvider) ForceFlush(ctx context.Context) error {
 func (p *TracerProvider) Shutdown(ctx context.Context) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if p.isShutdown {
+	if p.isShutdown.Load() {
```

Comment:
We can also add

```golang
if p.isShutdown.Load() {
  return
```

before aquiring the lock (`p.mu.Lock()`)

Then we will have the "book" implementation of double-checked locking 😉